### ISSUE TITLE
Nextcloud snap install breaks when password contains an exclamation point

### ISSUE THREAD

**Issue by winstonford** [nextcloud/nextcloud-snap#481]
### Nextcloud snap install breaks when password contains an exclamation point

It succeeds when password ends with an exclamation, albeit with the exclamation removed from the password.

1. sudo nextcloud.manual-install breaks when password contains an exclamation point
    - example: sudo nextcloud.manual-install test breaks!there
1. sudo nextcloud.manual-install succeeds when password ends in an exclamation point, but password does not contain exclamation point.
    - example: sudo nextcloud.manual-install test breaks!

---

### Creat Test Environment

1. Create do.co droplet NextcloudSnapTest
1. Ubuntu 16.04.4 32bit
1. 1gb mem
1. 1 vcpu
1. Add ssh key

---

### Install Nextcloud Snap with password containing exclamation point

1. https://www.digitalocean.com/community/tutorials/how-to-install-and-configure-nextcloud-on-ubuntu-16-04
1. Ssh in, adduser test, usermod -aG sudo test
1. sudo snap install nextcloud
1. sudo nextcloud.manual-install test breaks!there
1. Returns: -bash: !there: event not found

---

### Install Nextcloud Snap with password ending in exclamation point

1. sudo nextcloud.manual-install test breaks!
1. Returns: Nextcloud was successfully installed
1. Nextcloud is successfully installed with password: breaks

**Comment by kyrofa** [comment]
Regarding step 2, I just want to make sure you know there is absolutely zero correlation between the Linux user you're creating and the Nextcloud account you're creating in step 4. Step 4 can have its user named whatever you want, regardless of user accounts on the OS itself.

Okay, with that out of the way: this doesn't have anything to do with the snap, but with your shell. !<string> is a shell command that involves searching your history. You don't want your shell to interpret it, so try surrounding it with single quotes:

    $ sudo nextcloud.manual-install test 'breaks!there'

**Comment by kyrofa** [comment]
By the way: good bug report, thank you for all the info :smiley: .

**Comment by winstonford** [comment]
Regarding step 2, yes I learned that the linux user is different from the nextcloud user you are creating.. but when I first read the tut, I did actually think it was calling for my linux user 

And yes the shell!  I am not sure whether other shells use the same escapes as bash,  but perhaps I should suggest that the DO tutorial add the single quotes around the password for other n00bs using bash. 

Man, I am so rusty, thanks to your schooling I just remembered how I used to make notes in bash:
history | grep nextcloud > notes-nextcloud.txt

I used to type: # something 
to make an inline comment in my bash history about a previous cmd or output

Thanks!
